### PR TITLE
Bump 'NuGet.*' and 'Newtonsoft.Json' NuGet versions.

### PR DESCRIPTION
**Pull request by jpobst**
Update `NuGet.ProjectModel` to the latest version.  This new version requires a newer version of `Newtonsoft.Json`, so update to the latest of it as well.

Remove unneeded NuGet PackageReferences.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj [hunk 1]
```diff
@@ -28,9 +28,9 @@
     <PackageReference Include="Mono.Cecil" Version="$(MonoCecilVersion)" GeneratePathProperty="true" />
     <PackageReference Include="ILRepack" Version="2.0.18" />
     <PackageReference Include="Irony" />
-    <PackageReference Include="NuGet.Common" Version="6.7.0" />
-    <PackageReference Include="NuGet.Packaging" Version="6.7.0" />
-    <PackageReference Include="NuGet.ProjectModel" Version="6.7.0" />
+    <PackageReference Include="NuGet.Common" Version="6.9.1" />
+    <PackageReference Include="NuGet.Packaging" Version="6.9.1" />
+    <PackageReference Include="NuGet.ProjectModel" Version="6.9.1" />
```
[the remark this answers is not in the document]
Comment (jpobst) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj:33]: Yes, we started using them for Java binding dependency verification: https://github.com/xamarin/xamarin-android/pull/8649/files#diff-747215dcfa29e845d4d353aacccca25500165f49c03992ed7b789a7bdfc54031.

We use it to read the NuGet `project.assets.json` file.

[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]: Does something need to ship `NuGet.Packaging.dll`? I was looking for it in:

* https://github.com/xamarin/xamarin-android/blob/main/build-tools/installers/create-installers.targets

[the remark this answers is not in the document]
Comment (jpobst) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]: I think it gets ILLink'd into `Xamarin.Android.Build.Tasks.dll`.

[the remark this answers is not in the document]
Comment (jpobst) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]: There are tests for this feature: https://github.com/xamarin/xamarin-android/blob/main/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BindingBuildTest.cs#L775-L892

You may be right that it is finding the assemblies from another process.

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj [hunk 2]
```diff
@@ -28,9 +28,7 @@
     <PackageReference Include="Mono.Cecil" Version="$(MonoCecilVersion)" GeneratePathProperty="true" />
     <PackageReference Include="ILRepack" Version="2.0.18" />
     <PackageReference Include="Irony" />
-    <PackageReference Include="NuGet.Common" Version="6.7.0" />
-    <PackageReference Include="NuGet.Packaging" Version="6.7.0" />
-    <PackageReference Include="NuGet.ProjectModel" Version="6.7.0" />
+    <PackageReference Include="NuGet.Packaging" Version="6.9.1" />
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj:31]: One of these must be required:
```
/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs(15,13): error CS0234: The type or namespace name 'ProjectModel' does not exist in the namespace 'NuGet' (are you missing an assembly reference?) [/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]
/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs(398,52): error CS0246: The type or namespace name 'LockFileLibrary' could not be found (are you missing a using directive or an assembly reference?) [/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]
/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs(351,11): error CS0246: The type or namespace name 'LockFile' could not be found (are you missing a using directive or an assembly reference?) [/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]
/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs(356,29): error CS0246: The type or namespace name 'LockFile' could not be found (are you missing a using directive or an assembly reference?) [/Users/builder/azdo/_work/6/s/xamarin-android/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]
```

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**